### PR TITLE
Fix implicit-imports command not reporting implicit imports of external dependencies

### PR DESCRIPTION
**Pull request by rofle100lvl**
Resolves https://github.com/tuist/tuist/issues/6697

### Short description 📝

Added support for testing external dependencies on implicitness. Now instead of using allInternalTargets names as a mask for imports, using allTargets names.  

### How to test the changes locally 🧐

```
git clone https://github.com/karwa/tuist-external-implicit-dependency
tuist inspect implicit-imports tuist-external-implicit-dependency // Somewhere in xcode 

```
> Include a set of steps for the reviewer to test the changes locally (see [the documentation](https://docs.tuist.io/contributors/get-started) for reference).

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint-fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Comment by sphanley** [comment]
One thing I'm noticing when testing this against my project: my app's main target imports both Moya and RxMoya. In the Moya Package.swift, their RxMoya target defines a dependency on Moya. For reasons related to project structure, I'm opting to build Moya as a dynamic framework using the `productTypes` dictionary in my Pacakage.swift. This means that when I add `.external(name: "RxMoya")` to my app's main target, both the Moya.xcframework and RxMoya.xcframework are added with "Embed & Sign". However, this command still lists me as having an implicit dependency on Moya. Is that what you would expect?

**Comment by rofle100lvl** [comment]
> Is that what you would expect?
In my opinion and Marek opinion, you should not depend on products not declared in manifest(Package.swift). If you can make explicit dependency, you need to do it by yourself. Especially, if Moya and RXMoya declared as different
[https://github.com/Moya/Moya/blob/master/Package.swift](products). Am I right @fortmarek

**Comment by sphanley** [comment]
Thanks for the thoughts! That makes sense, I'm just newer to using SPM and wasn't sure whether this was intended to count as an implicit dependency, since the generated project _does_ end up with an explicit dependency on both the RxMoya and Moya targets simply by depending on RxMoya in the package.swift.

**Review by danieleformichelli — DISMISSED** [review]

File: Sources/TuistKit/Services/Inspect/InspecImplicitImportsService.swift [hunk 1]
```diff
@@ -60,23 +60,23 @@ final class InspectImplicitImportsService {
 
     private func lint(graphTraverser: GraphTraverser) async throws -> [InspectImplicitImportsServiceErrorIssue] {
         let allTargets = graphTraverser
+            .allTargets()
```
Comment (fortmarek) [inline Sources/TuistKit/Services/Inspect/InspecImplicitImportsService.swift:63]: nit: You can get `allTargets` by making a union of `allInternalTargets()` and `allExternalTargets()`, so we don't loop through some targets unnecessarily twice

**Review by fortmarek — COMMENTED** [review]
The implementation looks good 🙌 

Can you add some unit tests to test the new behavior?

**Review by fortmarek — DISMISSED** [review]
Thanks again for this fix!

**Review by fortmarek — APPROVED** [review]

**Pull request merged**